### PR TITLE
add sphinx docs

### PR DESCRIPTION
**Pull request by machow**
http://protowhat.readthedocs.io/en/docs/

should I remove the long paths? (e.g. `protowhat.checks.check_logic.test_or` to `test_or` ?)

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = sqlwhat
```
Comment (filipsch) [inline docs/Makefile]: protowhat

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+#
+# All configuration values have a default; values that are commented out
+# serve to show the default.
+
+# Mock heavy duty pythonwhat dependencies for autodoc
+import os
+import sys
+from unittest import mock
+ 
+MOCK_MODULES = ['pythonwhat', 'pythonwhat.Test', 'pythonwhat.Reporter', 'protowhat.grammar.plsql.ast']
```
Comment (filipsch) [inline docs/conf.py]: pythonwhat?

**Comment by machow** [comment]
It's a good question. If it's a doc on how to use a python library, then the full path makes sense. If it's a doc on how to write a datacamp SCT code block, then removing the path makes sense. Maybe we just leave it? I like the idea that people can just pop open the source code (even if it never happens ;).

**Comment by filipsch** [comment]
@machow both fair points :)

**Pull request merged**